### ISSUE TITLE
[HOLD for payment 2024-07-22] [HOLD for payment 2024-07-17] [regession] [$500] RBR transaction thread is disappearing from the LHN when navigating to another chat

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#36778]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.42-2
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @puneetlath 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1708116749863699

## Action Performed:

1. have an expense that you are the submitter on, which has violations
2. go to the transaction thread
3. notice that you see violations on the expense and that the RBR is shown in the LHN
4. navigate to a different chat

## Expected Result:
the RBR chat stays "pinned" in the LHN as long as the RBR is present

## Actual Result:
the chat disappears from the LHN

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/38435837/0d32661d-acc1-4c3e-b734-7fd93cdd9031

![Screenshot 2024-02-16 at 3 51 53 PM](https://github.com/Expensify/App/assets/38435837/3d21c73e-23b2-4c83-b7cd-8f65b1abf18d)

![Screenshot 2024-02-16 at 3 51 45 PM](https://github.com/Expensify/App/assets/38435837/c78f4fd7-97ff-4dc7-866d-cb0fc5bbfdaf)

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0198d221d75f59ca56</li>
        <li>Upwork Job ID: 1759384231310438400</li>
        <li>Last Price Increase: 2024-04-23</li>
<li>Automatic offers: </li>
<ul>
<li>jjcoffee | Reviewer | 0</li>
<li>tienifr | Contributor | 0</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @stephanieelliott / @stephanieelliott</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @stephanieelliott</details>

**Comment by youssef-lr** [comment]
This is getting fixed in https://github.com/Expensify/App/issues/33029. So I think we should hold this one.

**Comment by stephanieelliott** [comment]
Still on hold for  https://github.com/Expensify/App/issues/33029.

**Comment by stephanieelliott** [comment]
https://github.com/Expensify/App/issues/33029 is on prod now! @m-natarajan can you please retest for us?

**Comment by mvtglobally** [comment]
Issue is reproducible during KI retests.
https://platform.applause.com/services/links/v1/external/0f18d29bd4aff3922804c80bcae5f2205e699b8b5f9c91f36d659bc6d54f458f

**Comment by jjcoffee** [comment]
I guess we can re-hold on https://github.com/Expensify/App/issues/33029 since the PR was reverted due to performance issues?

**Comment by stephanieelliott** [comment]
https://github.com/Expensify/App/issues/33029 is still WIP, holding.

**Comment by stephanieelliott** [comment]
Still held on https://github.com/Expensify/App/issues/33029

**Comment by stephanieelliott** [comment]
Still held on https://github.com/Expensify/App/issues/33029

**Comment by jjcoffee** [comment]
@stephanieelliott There doesn't seem to be much progress on that other issue, so I wonder if it's worth us just proceeding here?

**Comment by stephanieelliott** [comment]
Yeah @jjcoffee I think we can proceed -- the hold was originally because we thought https://github.com/Expensify/App/issues/33029 would fix this, but we retested for the short time it was on prod and it was still occurring. So I think we can assume this issue needs it's own fix.

**Comment by stephanieelliott** [comment]
Reapplying the `Bug` label to get another BZ member on this while I am OOO til May 2. Thanks @mallenexpensify -- to catch you up, we just took this issue off hold to get more proposals. I'll pick this one back up from you if it is still open when I come back next week!

**Comment by jjcoffee** [comment]
@tienifr It's been a while since you made your [proposal](https://github.com/Expensify/App/issues/36778#issuecomment-1951585989), can you confirm that it still applies to the latest main?

**Comment by jjcoffee** [comment]
Friendly bump @tienifr, otherwise open to other proposals!

**Comment by jjcoffee** [comment]
@tienifr Thanks for the update! I like the overall approach, but did you happen to run any performance tests? Seems like the other PR failed because of performance issues, so we'd want to avoid that here.

**Comment by jjcoffee** [comment]
@tienifr Perfect, thanks! Let's give your [proposal](https://github.com/Expensify/App/issues/36778#issuecomment-1951585989) a try!

:ribbon::eyes::ribbon: C+ reviewed

**Comment by jjcoffee** [comment]
@tienifr Okay, just a heads up I'll be OOO 6-13th (I may have some time on the 6th).

**Comment by stephanieelliott** [comment]
I'm back from OOO, thanks for watching this while I was out @mallenexpensify!

**Comment by stephanieelliott** [comment]
PR is under active review/discussion

**Comment by jjcoffee** [comment]
Reviewing the PR again today!

**Comment by jjcoffee** [comment]
PR reviewed, just one potential tweak to the sorting (if it doesn't affect performance).

**Comment by stephanieelliott** [comment]
Comments on PR are resolved, just waiting to be merged

**Comment by stephanieelliott** [comment]
This was merged to main

**Comment by jjcoffee** [comment]
New PR under review here https://github.com/Expensify/App/pull/43502

**Comment by stephanieelliott** [comment]
This caused a regression for which the fix PR is still open. Removing payment info as we'll base payment on when the new PR is merged! https://github.com/Expensify/App/pull/43502

**Comment by jjcoffee** [comment]
Just a heads up that I'll be going OOO until Jul 24 or so, but the PR is now approved by me and I'll be able to check in now and then if anything comes up that needs handling.

**Comment by stephanieelliott** [comment]
Fix PR was merged to main earlier today!

**Comment by stephanieelliott** [comment]
Applying the `New Feature` label to get another BZ member on this while I am OOO til July 10. To catch you up on where we are @puneetlath: this issue caused a regression which was fixed by PR https://github.com/Expensify/App/pull/43502. That PR should be deployed to staging soon -- once it hits prod you'll want to update this issue's title with the new payment date. 

Thanks for watching this, I'll grab this back from you when I return!

**Comment by mallenexpensify** [comment]
@stephanieelliott , @tienifr won't be paid on this issue due to a double payment on https://github.com/Expensify/App/issues/38054#issuecomment-2219286959

**Comment by mallenexpensify** [comment]
Contributor: @tienifr not paid $500 due to a double payment on  https://github.com/Expensify/App/issues/38054#issuecomment-2219286959
Contributor+: @jjcoffee paid $500 via Upwork. 

@jjcoffee can you fill out the BZ checklist above? thx

**Comment by jjcoffee** [comment]
- The PR that introduced the bug has been identified. Link to the PR: N/A - this was just never implemented
- The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: N/A
- A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: N/A
- Determine if we should create a regression test for this bug. **Yes** - we have multiple we need to add
- If we decide to create a [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) for the bug, please propose the regression test steps to ensure the same bug will not reach production again.

**Regression Test Proposal**

**Test 1**

1. Have an expense that you are the submitter on, which has violations
2. Go to the transaction thread
3. Notice that you see violations on the expense and that the RBR is shown in the LHN
4. Navigate to a different chat
5. Verify the RBR chat remains in the LHN as long as the RBR is present

**Test 2**

1. Go to a workspace chat that has no unsettled expense.
2. Create an expense that will induce RBR (e.g. on a workspace where category is required submit an expense without category).
3. Click on the report with RBR on LHN.
4. Click on the header subtitle.
5. Verify app will return to the workspace main chat.

**Test 3**

Prerequisite: User A and B have no chatted before and both accounts are new.
1. [User A] Go offline
2. [User A] Submit an expense to User B
3. [User B] Submit an expense with the same amount to User A
4. [User A] Go online
5. Verify user A will not have two same transaction threads with RBR on LHN.

Do we agree 👍 or 👎

**Comment by stephanieelliott** [comment]
Thanks @jjcoffee! Issue for test case here: https://github.com/Expensify/Expensify/issues/415683
